src/CRM/CivixBundle/Resources/views/Code/api4.php.php

```diff
@@ -0,0 +1,47 @@
+<?php echo "<?php\n"; ?>
+namespace Civi\Api4;
+
+use Civi\Api4\Generic\AbstractAction;
+use Civi\Api4\Generic\Result;
+use Civi\Api4\Generic\BasicGetFieldsAction;
+
+/**
+*  Class <?php echo $entityNameCamel ?>.
+*
+* Provided by the <?php echo $extensionName ?> extension.
+*
+* @package Civi\Api4
+*/
+class <?php echo $entityNameCamel ?> extends Generic\AbstractEntity {
```

Comment:
I rarely recommend extending the `AbstractEntity` class directly, as there are more useful base classes.
If the entity in question has a DAO, then it definitely should use `DAOEntity`. But Civix already generates an APIv4 `DAOEntity` whenever generating an entity, so presumably you're targeting a different use-case?

`BasicEntity` is the go-to for any type of CRUD API that isn't backed by a DAO. Is that what you're after?